src/docs/arch-block.dox

```diff
@@ -172,18 +172,53 @@ The block manager doesn't guarantee that calling \c bm->write will result in
 the data being flushed to disk. In the checkpoint scenario WiredTiger will also
 call \c bm->sync which will call the file system dependent flush function.
 
-As a checkpoint progresses it updates its three extent lists, these extent lists
-are written out to disk as part of the checkpoint in blocks. When a checkpoint
-is written the previous checkpoints will be cleaned up if possible, suppose
-the block manager has finished writing out a checkpoint. Which we will call
-checkpoint \c C, checkpoints \c A and \c B exist on disk. By default the block
-manager only retains two checkpoints so it can cleanup checkpoint \c A.
-Checkpoint \c A's extent lists get merged into checkpoint \c B's extent lists.
-If a block, \c N, is created in checkpoint \c A and removed in checkpoint \c B
-the space occupied by the block is considered free and can be utilized.
+* Checkpoint deletion and merging *
+
+As a checkpoint progresses it takes a snapshot of the three extent lists kept
+by the block manager, these extent lists are written out to disk as part of
+the checkpoint in blocks. Between checkpoints these extent lists are being
+updated via normal operation of WiredTiger.
+
+Suppose we have a checkpoint \c A, which has an alloc list which contains 3
+blocks \c I, \c J, \c K as such its extent lists are as follows:
+
+Alloc: \c I, \c J, \c K
+Avail: Empty
```

Comment:
It would help to show L on the Avail list here.  I realize we might get L by growing the file, but it will illustrate the use of extent lists better to show L being allocated from Avail.